src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineParser.java

```diff
@@ -181,11 +182,24 @@ public String getVersion() {
     public void usage(final PrintStream stream, final boolean printCommon) {
         stream.print(getStandardUsagePreamble(callerArguments.getClass()) + getUsagePreamble());
         stream.println("\n" + getVersion());
-        stream.println("\n\nArguments:\n");
+        stream.println("\n\nRequired Arguments:\n");
+
+        // print required args first; temporarily buffer output for optional args
```

Comment:
This business with a second output stream seems overly clever.  I think just explicitly printing one and then the other would be much easier to read.

Something like this:
```
   List<ArgumentDefinition> args = argumentDefinitions.stream()
                .filter(argumentDefinition -> printCommon || !argumentDefinition.isCommon)
                .collect(Collectors.toList());

      stream.println("\n\nRequired Arguments:\n");
        args.stream().filter(a -> !a.optional).forEachOrdered(a -> printArgumentUsage(stream, a));

        stream.println("Optional Arguments");
        args.stream().filter(a -> a.optional).forEachOrdered(a -> printArgumentUsage(stream, a));
```